netty/src/main/java/io/grpc/netty/NettyClientStream.java

```diff
@@ -50,12 +50,25 @@
 import io.netty.handler.codec.http2.Http2Stream;
 import io.netty.util.AsciiString;
 
+import java.util.IdentityHashMap;
+import java.util.Map;
+import java.util.function.Supplier;
```

Comment:
After thinking about this over a long bike ride, I think I am going to revert this change in favor of a cleaner solution.  Sorry for the wasted review.